src/frontend/src/helpers/Login.ts

```diff
@@ -0,0 +1,59 @@
+import * as $ from "jquery";
+export {Login, LoginError, Connection}
+
+class LoginError extends Error {
+  constructor(msg: string,
+              public username?: string,
+              public status?: number,
+              public statusText?: number) {
+    super(msg);
+  }
+}
+
+class Login {
```

Comment:
Well the thing is everything you're dealing with here is global data.  I mean, I don't care if it's just a class with static methods or just global functions, but traditionally we've just used global functions.